Implement all targets to avoid msbuild p2p problems

[This file](https://github.com/dotnet/Nerdbank.GitVersioning/blob/master/src/Nerdbank.GitVersioning.Tasks/build/PrivateP2PCaching.proj) should define all [these targets](https://github.com/dotnet/msbuild/blob/main/documentation/ProjectReference-Protocol.md#targets-required-to-be-referenceable) to avoid regressions when people update from 3.4 to 3.5 of NB.GV.

FWIW, I just ran into `C:\Users\alex\source\repos\system-terminal\pkg\cache\nerdbank.gitversioning\3.5.50-alpha\build\PrivateP2PCaching.proj : error MSB4057: The target "Publish" does not exist in the project.` when invoking `dotnet publish`.

--- a/src/Nerdbank.GitVersioning.Tasks/build/PrivateP2PCaching.proj
+++ b/src/Nerdbank.GitVersioning.Tasks/build/PrivateP2PCaching.proj
@@ -13,6 +13,7 @@
   <!-- This target is called by Clean or Rebuild on a referencing project. -->
   <Target Name="Clean" />
 
-  <!-- This target is called by Pack on a referencing project. -->
+  <!-- This target is called by their respective targets on referencing projects. -->
   <Target Name="Pack" />
+  <Target Name="Publish" />
 </Project>

